modules/adopter-statistics-impl/src/main/java/org/opencastproject/adopter/statistic/Sender.java

```diff
@@ -0,0 +1,137 @@
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Contains methods for sending statistic data via rest.
+ */
+public class Sender {
+
+  /** The logger */
+  private static final Logger logger = LoggerFactory.getLogger(Sender.class);
+
+  /** The base URL for the external server where the data will be send to */
+  private String baseUrl;
+
+  // The suffixes for the base statistic server URL
+  // they determine to which REST endpoint the data will be sent
+  private static final String GENERAL_DATA_URL_SUFFIX = "adopter";
+  private static final String STATISTIC_URL_SUFFIX = "statistic";
+
+  //================================================================================
+  // Constructor
+  //================================================================================
+
+  /**
+   * Simple Constructor that requires the URL of the statistic server.
+   * @param statisticServerBaseUrl The URL prefix of the statistic server.
+   */
+  public Sender(String statisticServerBaseUrl) {
+    if (!statisticServerBaseUrl.endsWith("/")) {
+      statisticServerBaseUrl += "/";
+    }
+    this.baseUrl = statisticServerBaseUrl;
+  }
+
+  //================================================================================
+  // Methods
+  //================================================================================
+
+  /**
+   * Executes the 'send' method with the proper REST URL suffix.
+   * @param json The data which shall be sent.
+   * @throws Exception General exception that can occur while sending the data.
+   */
+  public void sendGeneralData(String json) throws Exception {
+    send(json, GENERAL_DATA_URL_SUFFIX);
+  }
+
+  /**
+   * Executes the 'send' method with the proper REST URL suffix.
+   * @param json The data which shall be sent.
+   * @throws Exception General exception that can occur while sending the data.
+   */
+  public void sendStatistics(String json) throws Exception {
+    send(json, STATISTIC_URL_SUFFIX);
+  }
+
+  /**
+   * Sends the JSON string via post request.
+   * @param json The JSON string that has to be send.
+   * @param urlSuffix The url suffix determines to which rest endpoint the data will be send.
+   * @throws Exception General exception that can occur while processing the POST request.
+   */
+  private void send(String json, String urlSuffix) throws Exception {
+    try {
+      URL url = new URL(baseUrl + urlSuffix);
+      HttpURLConnection con = (HttpURLConnection) url.openConnection();
+      con.setRequestMethod("POST");
+      con.setRequestProperty("Content-Type", "application/json; utf-8");
+      con.setRequestProperty("Accept", "application/json");
+      con.setDoOutput(true);
+
+      try (OutputStream os = con.getOutputStream()) {
+        byte[] input = json.getBytes(StandardCharsets.UTF_8);
+        os.write(input, 0, input.length);
+      }
+
+      String httpStatus = con.getResponseCode() + "";
```

Comment:
Let's replace this logic with something like

`boolean errorOccurred = con.getResponseCode() < 200 || con.getResponseCode() >= 300`